hyperspy/_components/power_law.py

```diff
@@ -109,12 +109,14 @@ def estimate_parameters(self, signal, x1, x2, only_current=False,
             s = signal.get_current_signal()
         else:
             s = signal
-        I1 = s.isig[i1:i3].integrate1D(2j).data.astype("float")
-        I2 = s.isig[i3:i2].integrate1D(2j).data.astype("float")
         if s._lazy:
             import dask.array as da
             log = da.log
+            I1 = s.isig[i1:i3].integrate1D(2j).data.astype("float")
+            I2 = s.isig[i3:i2].integrate1D(2j).data.astype("float")
         else:
+            I1 = s.data[..., i1:i3].sum(-1)
```

Comment:
This is an speed improvement of 100x. Using `integrate1D` give slightly different results (~2e-3) for binned signal, so for an estimation this is fine!

Reply:
I don't get it: it should give exactly the same result for binned signals and be a worst approximation for unbinned signals.

Reply:
After looking at the tests, that's probably what you meant.